doc/how-to-package-haskell-code.rst

```diff
@@ -0,0 +1,291 @@
+How to package Haskell code
+===========================
+
+.. TIP::
+    If this is your first time using `cabal` you should check out the :doc:`Getting Started guide <getting-started>`.
+
+Starting from scratch, we're going to walk you through creating a simple
+Haskell application.
+
+**TL;DR;** ``mkdir proglet && cd proglet && cabal init --simple --exe && cabal run proglet``
+
+
+Introduction
+------------
+
+Every application needs a name, we'll call ours "proglet" and start by
+creating an empty directory.
+
+.. highlight:: console
+
+::
+
+    $ mkdir proglet
+    $ cd proglet/
+
+
+.. _init quickstart:
+
+Using ``cabal init``
+--------------------
+
+The ``cabal init`` command creates the necessary files for a Cabal package,
+it has both an ``--interactive`` (default) and ``--non-interactive``
+mode. The interactive mode will walk you through many of the package
+options and metadata, the non-interactive mode will simply pick reasonable
+defaults which is sufficient if you're just trying something out.
+
+.. highlight:: console
+
+::
+
+    $ cabal init --non-interactive
+    # You can also use -n which is the short version of --non-interactive
+
+If you want, you can also try out the interactive mode, for now chose
+"Executable" when asked what type of package you want to build.
+
+.. highlight:: console
+
+::
+
+    $ cabal init
+    ...
+    What does the package build:
+       1) Executable
+       2) Library
+       3) Library and Executable
+       4) Test suite
+    Your choice?
+
+One of the important questions is whether the package contains a library
+and/or an executable. Libraries are collections of Haskell modules that
+can be re-used by other Haskell libraries and programs, while executables
+are standalone programs. Test suites can both depend on a library or be
+standalonely generated.
+
+For the moment these are the only choices. For more complex packages
+(e.g. a library and multiple executables) the ``.cabal``
+file can be edited afterwards.
+
+After you make your selection (executable; library; library
+and executable; or: test suite) cabal asks us a number of questions starting with
+which version of the cabal specification to use, our package's name
```

Comment:
```suggestion
which version of the cabal specification to use, your package's name
```
Alternately change "after you make your selection" above; these should match instead of suddenly switching from second to first person.